src/assets/scripts/client/measurement/MeasureLegModel.js

```diff
@@ -0,0 +1,277 @@
+         * The end point of the leg
+         *
+         * @for MeasureLegModel
+         * @property endPoint
+         * @type {array<number>}
+         * @default endPoint
+         */
+        this.endPoint = endPoint;
+
+        /**
+         * The length of the leg (in km)
+         *
+         * @for MeasureLegModel
+         * @property _distance
+         * @type {number}
+         * @default null
+         * @private
+         */
+        this._distance = null;
+
+        /**
+         * The list of text labels to be displayed
+         *
+         * @for MeasureLegModel
+         * @property labels
+         * @type {labels}
+         * @default null;
+         */
+        this.labels = null;
+
+        /**
+         * The end point of the leg
+         *
+         * @for MeasureLegModel
+         * @property _midPoint
+         * @type {array<number>|null}
+         * @default null
+         * @private
+         */
+        this._midPoint = null;
+
+        /**
+         * The midpoint of this leg
+         *
+         * @for MeasureLegModel
+         * @property _next
+         * @type {MeasureLegModel}
+         * @default null;
+         * @private
+         */
+        this._next = null;
+
+        /**
+         * The previous leg
+         *
+         * @for MeasureLegModel
+         * @property _previous
+         * @type {MeasureLegModel}
+         * @default previousLeg
+         * @private
+         */
+        this._previous = previousLeg;
+
+        /**
+         * The turn radius (in km)
+         *
+         * @for MeasureLegModel
+         * @property _radius
+         * @type {number}
+         * @default radius
+         * @private
+         */
+        this._radius = radius;
+
+        this._init(previousLeg);
+    }
+
+    /**
+     * The bearing along the leg (in radians)
+     *
+     * @for MeasureLegModel
+     * @property bearing
+     * @type {number}
+     */
+    get bearing() {
+        return this._bearing;
+    }
+
+    /**
+     * The length of the leg (in km)
+     *
+     * @for MeasureLegModel
+     * @property distance
+     * @type {number}
+     */
+    get distance() {
+        return this._distance;
+    }
+
+    /**
+     * The midpoint of this leg
+     *
+     * @for MeasureLegModel
+     * @property midPoint
+     * @type {array<number>|null}
+     */
+    get midPoint() {
+        return this._midPoint;
+    }
+
+    /**
+     * The next leg
+     *
+     * @for MeasureLegModel
+     * @property next
+     * @type {MeasureLegModel}
+     */
+    get next() {
+        return this._next;
+    }
+
+    /**
+     * The previous leg
+     *
+     * @for MeasureLegModel
+     * @property next
+     * @type {MeasureLegModel}
+     */
+    get previous() {
+        return this._previous;
+    }
+
+    /**
+     * The turn radius (in km)
+     *
+     * @for MeasureLegModel
+     * @property _radius
+     * @type {number}
+     */
+    get radius() {
+        const response = this._radius;
```

Comment:
what does `response` mean here?  feels like a strange word in relation to `radius` an measuring.

Reply:
Don't judge me, but this is a side effect from my Basic days when I didn't like to name a local variable the same as the method it's declared in. :duck:
It sometimes leaks through...